org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/preferences/LifecycleMappingsViewer.java

```diff
@@ -444,7 +444,8 @@ String getSourcelabel(MojoExecutionKey execution, List<IPluginExecutionMetadata>
     LinkedHashSet<String> sources = new LinkedHashSet<>();
     if(mappings != null && !mappings.isEmpty()) {
       for(IPluginExecutionMetadata mapping : mappings) {
-        if(mapping instanceof LifecycleMappingMetadataSource metadata) {
+        LifecycleMappingMetadataSource metadata = ((PluginExecutionMetadata) mapping).getSource();
```

Comment:
Either use `IPluginExecutionMetadata` (without a cast) or add a check that mapping is actually of type `PluginExecutionMetadata` ...